src/primitive_debug.cc

```diff
@@ -60,9 +54,40 @@ PRIMITIVE(object_histogram) {
     } else if (is_string(object) && !String::cast(object)->content_on_heap()) {
       size += String::cast(object)->length() + 1;
     }
-    capture.data[class_index * UINT32_PER_ENTRY + 0] += 1;
-    capture.data[class_index * UINT32_PER_ENTRY + 1] += size;
+    data[class_index].count++;
+    data[class_index].size += size;
   });
+  int non_trivial_entries = 0;
+  uword bytes_needed = 0;
+  for (int i = 0; i < length; i++) {
+    if (data[i].size > 0) {
+      non_trivial_entries++;
+      bytes_needed += Encoder::bytes_to_encode(i);
+      bytes_needed += Encoder::bytes_to_encode(data[i].count);
+      bytes_needed += Encoder::bytes_to_encode(data[i].size);
+    }
+  }
+  MallocedBuffer encoding_buffer(100 + bytes_needed);
+  if (!encoding_buffer.has_content()) MALLOC_FAILED;
+
+  ProgramOrientedEncoder encoder(program, &encoding_buffer);
+
+  encoder.write_header(non_trivial_entries * 3, 'O');  // O for objects.  See mirror.toit.
```

Comment:
I think I'd move this to the Encoder implementation to make it consistent with the other high-level encoder methods.

Reply:
This is done in the same way as profiles.  It feels strange to put something so specific in a generic place like encoder.cc